src/Router/buildClientApp.tsx

```diff
@@ -0,0 +1,77 @@
+import React from "react"
+import BrowserProtocol from "farce/lib/BrowserProtocol"
+import HashProtocol from "farce/lib/HashProtocol"
+import MemoryProtocol from "farce/lib/MemoryProtocol"
+import createInitialFarceRouter from "found/lib/createInitialFarceRouter"
+import createRender from "found/lib/createRender"
+import queryMiddleware from "farce/lib/queryMiddleware"
+import { AppShell } from "./AppShell"
+import { Resolver } from "found-relay"
+import { createEnvironment } from "../Relay/createEnvironment"
+import { loadComponents } from "loadable-components"
+import { data as sd } from "sharify"
+import { AppConfig, ClientResolveProps } from "./types"
+
+export function buildClientApp(config: AppConfig): Promise<ClientResolveProps> {
+  return new Promise(async (resolve, reject) => {
+    try {
+      const { routes, historyProtocol = "browser" } = config
+      const relayBootstrap = JSON.parse(window.__RELAY_BOOTSTRAP__ || "{}")
+
+      const relayEnvironment = createEnvironment({
+        cache: relayBootstrap,
+        user: sd.CURRENT_USER,
+      })
+
+      const getHistoryProtocol = () => {
+        switch (historyProtocol) {
+          case "browser":
+            return new BrowserProtocol()
+          case "hash":
+            return new HashProtocol()
+          case "memory":
+            return new MemoryProtocol("/")
+          default:
+            return new BrowserProtocol()
+        }
+      }
```

Comment:
Just out of curiosity, why did you make this a function?

Reply:
Just for list readability, like 

```javascript
const getHistoryProtocol = () => {
  switch (historyProtocol) {
    case "browser":
      return new BrowserProtocol()
    case "hash":
      return new HashProtocol()
    case "memory":
      return new MemoryProtocol("/")
    default:
      return new BrowserProtocol()
  }
}
```

reads better to my eye than 

```javascript 
if (historyProtocol === 'browser') {
  return new BrowserProtocol()
} else if (historyProtocol === 'hash') {
  return new HashProtocol()
} else if (historyProtocol === 'memory') {
  return new BrowserProtocol()
}
```